src/dash-table/derived/edges/index.ts

```diff
@@ -1,27 +1,30 @@
-import * as R from 'ramda';
-
 import { BorderStyle, BORDER_PROPERTIES } from './type';
 import { IConvertedStyle } from '../style';
 import { Datum, IVisibleColumn } from 'dash-table/components/Table/props';
 import { matchesDataCell, matchesDataOpCell, matchesFilterCell, getFilterOpStyles, matchesHeaderCell, getHeaderOpStyles } from 'dash-table/conditional';
+import { traverse2 } from 'core/math/matrixZipMap';
 
 function resolveEdges(styles: IConvertedStyle[]): BorderStyle {
     let res: BorderStyle = {};
 
-    R.addIndex<IConvertedStyle>(R.forEach)((s, i) => R.forEach(p => {
-        const border = s.style[p] || s.style.border;
+    traverse2(
+        styles,
+        BORDER_PROPERTIES,
+        (s, p, i) => {
+            const border = s.style[p] || s.style.border;
 
-        if (border) {
-            res[p] = [border, i];
+            if (border) {
+                res[p] = [border, i];
+            }
         }
-    }, BORDER_PROPERTIES), styles);
+    );
 
     return res;
 }
 
-export const getDataCellEdges = (datum: Datum, i: number, column: IVisibleColumn) => R.compose(resolveEdges, matchesDataCell(datum, i, column));
-export const getDataOpCellEdges = (datum: Datum, i: number) => R.compose(resolveEdges, matchesDataOpCell(datum, i));
-export const getFilterCellEdges = (column: IVisibleColumn) => R.compose(resolveEdges, matchesFilterCell(column));
-export const getFilterOpCellEdges = () => R.compose(resolveEdges, getFilterOpStyles);
-export const getHeaderCellEdges = (i: number, column: IVisibleColumn) => R.compose(resolveEdges, matchesHeaderCell(i, column));
-export const getHeaderOpCellEdges = (i: number) => R.compose(resolveEdges, getHeaderOpStyles(i));
\ No newline at end of file
+export const getDataCellEdges = (datum: Datum, i: number, column: IVisibleColumn) => (styles: IConvertedStyle[]) => resolveEdges(matchesDataCell(datum, i, column)(styles));
+export const getDataOpCellEdges = (datum: Datum, i: number) => (styles: IConvertedStyle[]) => resolveEdges(matchesDataOpCell(datum, i)(styles));
+export const getFilterCellEdges = (column: IVisibleColumn) => (styles: IConvertedStyle[]) => resolveEdges(matchesFilterCell(column)(styles));
+export const getFilterOpCellEdges = () => (styles: IConvertedStyle[]) => resolveEdges(getFilterOpStyles(styles));
+export const getHeaderCellEdges = (i: number, column: IVisibleColumn) => (styles: IConvertedStyle[]) => resolveEdges(matchesHeaderCell(i, column)(styles));
+export const getHeaderOpCellEdges = (i: number) => (styles: IConvertedStyle[]) => resolveEdges(getHeaderOpStyles(i)(styles));
```

Comment:
ha good call - `compose` when you're not currying seems particularly silly to me.